tests/unit/_autoscaling/test_autoscaled_pool.py

```diff
@@ -208,6 +209,60 @@ def get_historical_system_info() -> SystemInfo:
             await pool_run_task
 
 
+async def test_autoscales_uses_desired_concurrency_ratio(system_status: SystemStatus | Mock) -> None:
+    """Test that desired concurrency ratio can limit desired concurrency.
+
+    This test crates situation where only one task is ready and then no other task is ever ready.
+    This creates situation where the system could scale up desired concurrency, but it will not do so because
+    desired_concurrency_ratio=1 means that first the system would have to increase current concurrency to same number as
+    desired concurrency and due to no other task ever being ready, it will never happen. Thus desired concurrency will
+    stay 2 as was the initial setup, even though other conditions would allow the increase. (max_concurrency=4,
+    system being idle)."""
+
+    async def run() -> None:
+        await asyncio.sleep(0.1)
+
+    is_task_ready_iterator = chain([future(True)], repeat(future(False)))
+
+    def is_task_ready_function() -> Awaitable[bool]:
+        return next(is_task_ready_iterator)
```

Comment:
I know that `AutoscaledPool` requires this to be awaitable, but shouldn't it then be defined as async?

```suggestion
    async def is_task_ready_function() -> Awaitable[bool]:
        return next(is_task_ready_iterator)
```

Reply:
No, but I was thinking the same when I first saw existing tests in this file.
This function returns [future](https://github.com/apify/crawlee-python/blob/27db2e4162324f40a3b5059dbc40f66d47843876/tests/unit/_autoscaling/test_autoscaled_pool.py#L30). So it is normal function, but the return value is awaitable. (Exactly what type hints require: `is_task_ready_function: Callable[[], Awaitable[bool]]`)

Reply:
Ah, thanks for the explanation.